### PR TITLE
Force reanalysis of rhnChannelPackage earlier (bsc#1186704)

### PR DESCRIPTION
**Pull request by rjmateus**
refactoring to run tables analyze in CLM

Signed-off-by: Ricardo Mateus <rmateus@suse.com>

## What does this PR change?

During build validation tests a CLM project was created to flat centos repository. The build action of that project cause Postgres to be using 100% CPU and got stuck in one specific query. Run that query outside CLM process was really fast and uses all expected indexes.
As a side-effect, Uyuni server got completely unusable, and UI unresponsible.

This PR aims to solve this performance problem by running a database table analyze in the more used CLM tables during build.

Fix For: https://bugzilla.suse.com/show_bug.cgi?id=1186704

## GUI diff

No difference.

Before:

After:

- [ ] **DONE**

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- No documentation needed: only internal and user invisible changes
- Documentation issue was created: [Link for SUSE Manager contributors](https://github.com/SUSE/spacewalk/issues/new?template=ISSUE_TEMPLATE_DOCUMENTATION.md&labels=documentation&projects=SUSE/spacewalk/31), [Link for community contributors](https://github.com/uyuni-project/uyuni-docs/issues/new).
- (OPTIONAL) [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pulls)

- [ ] **DONE**

## Test coverage
- No tests: **add explanation**
- No tests: already covered
- Unit tests were added
- Cucumber tests were added

- [ ] **DONE**

## Links

Fixes: https://github.com/SUSE/spacewalk/issues/15070

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: java/code/src/com/redhat/rhn/manager/contentmgmt/ContentManager.java [hunk 1]
```diff
@@ -980,6 +987,19 @@ public void alignEnvironmentTargetSync(Collection<ContentFilter> filters, Channe
         ChannelManager.queueChannelChange(tgt.getLabel(), "java::alignChannel", "Channel aligned");
     }
 
+    /**
+     * Run database analyze in tables more affected by the CLM channel align
+     */
+    private void analyzeAlignTables(){
+        // a lot was inserted into tables at this point. Make sure stats are up-to-date before continuing
```
Comment (aaannz) [inline java/code/src/com/redhat/rhn/manager/contentmgmt/ContentManager.java:994]: Nitpick, I think this comment made more sense in the original location rather then in helper.

**Review by aaannz — APPROVED** [review]
👍🏾

**Review by juliogonzalez — APPROVED** [review]
Changelog and target branch look good, but review the failed checkstyle test :-)

File: java/code/src/com/redhat/rhn/manager/contentmgmt/ContentManager.java (hunk 1 above)
Comment (rjmateus) [reply java/code/src/com/redhat/rhn/manager/contentmgmt/ContentManager.java]: comment moved to the original location

**Review by moio — APPROVED** [review]
Good from my side. I guess this is covered by build validation tests (please fill that in the PR description).

It would be great to have a **Speedup:** comment (even with coarse numbers) to record the patch impact for the posterity :wink:

**Comment by rjmateus** [comment]
Speedup improvements in the Builv Validation environment:
Before Patch: build run for more than 4 hours without finishing and blocking the server, making it unusable
After patch: Build run in 14 minutes, without any query bloked. The server was slower but usable.

**Pull request merged**